internal/operands/common-templates/reconcile.go

```diff
@@ -171,7 +180,8 @@ func reconcileOlderTemplates(request *common.Request) ([]common.ReconcileFunc, e
 	for i := range existingTemplates.Items {
 		template := &existingTemplates.Items[i]
 		funcs = append(funcs, func(*common.Request) (common.ResourceStatus, error) {
-			return common.CreateOrUpdateClusterResource(request, template,
+			return common.CreateOrUpdateClusterResource(request,
```

Comment:
@akrejcir what would you think about adding
`CreateOrUpdateClusterResourceWithAppLabels(request *Request, name string, component AppComponent, resource controllerutil.Object, updateResource ResourceUpdateFunc)`
which would move the need for those AddAppLabels calls into a wrapped function call?
I added it locally and it looks good so far.

Reply:
So it would become
```suggestion
			return common.CreateOrUpdateClusterResourceWithAppLabels(request, operandName, operandComponent, template,
```

Reply:
I think we need to create a builder for these calls. Then each resource can use what it needs. For example:
```
return common.CreateOrUpdate(request, resource).
    ClusterResource().
    WithAppLabels(name, component).
    UpdateFunc(...).
    Execute()
```

Let's leave it for another PR.